test/core/util/test_config.cc

```diff
@@ -135,26 +97,42 @@ static void print_stack_from_context(CONTEXT c) {
 #endif
 
   HANDLE process = GetCurrentProcess();
-  HANDLE thread = GetCurrentThread();
 
   SYMBOL_INFOW* symbol =
       (SYMBOL_INFOW*)calloc(sizeof(SYMBOL_INFOW) + 256 * sizeof(wchar_t), 1);
   symbol->MaxNameLen = 255;
   symbol->SizeOfStruct = sizeof(SYMBOL_INFOW);
 
-  while (StackWalk(imageType, process, thread, &s, &c, 0,
-                   SymFunctionTableAccess, SymGetModuleBase, 0)) {
-    BOOL has_symbol =
-        SymFromAddrW(process, (DWORD64)(s.AddrPC.Offset), 0, symbol);
-    fwprintf(
-        stderr, L"*** %016I64X %ls - %016I64X\n", (DWORD64)(s.AddrPC.Offset),
-        has_symbol ? symbol->Name : L"<<no symbol>>", (DWORD64)symbol->Address);
+  const unsigned short MAX_CALLERS_SHOWN = 32;
+  for (int frame = 0; frame < MAX_CALLERS_SHOWN && StackWalk(imageType, process, thread, &s, &c, 0,
```

Comment:
Looks like before, the abort handler would limit the number of stack frames to display up to `MAX_CALLERS_SHOWN`, but the exception/crash handler would not, and would continue walking the stack as long as `StackWalk` returned true.

Is it necessary to limit the number of frames displayed to `MAX_CALLERS_SHOWN` in both cases? If not, maybe pass the limit on number of frames to display as a parameter?

Reply:
I think the previous limit was due to the fact that the method to grab the stacktrace had a hard limit on number of frames.
This limitation no longer exists, so I set the print limit to 8192 frames (seems pretty high) to avoid completely flooding the log if things go haywire for some reason.